cocotb/handle.py

```diff
@@ -755,6 +760,11 @@ def _set_value(self, value, call_sim):
                 DeprecationWarning, stacklevel=3)
             value = BinaryValue(value=cocotb.utils.pack(value), n_bits=len(self))
         elif isinstance(value, dict):
+            warnings.warn(
+                "dict values are no longer accepted for value assignment. "
```

Comment:
Wording is a bit odd, since the dict is still accepted, just with a warning

Reply:
Following the wording from the `ctypes.Structure` example above. You have a suggestion? I'll probably change the other one as well.